src/map/pc.c

```diff
@@ -6829,6 +6831,16 @@ static void pc_calcexp(struct map_session_data *sd, uint64 *base_exp, uint64 *jo
 		}
 #endif
 
+#ifdef RENEWAL
+		// after rebalance, boss monsters does give more EXP, but "MVP" exp is not increased
+		if (sd->sc.data[SC_RICHMANKIM] != NULL && src->type == BL_MOB && (flags & EXP_FLAG_MVP) == 0) {
+			buff_ratio += sd->sc.data[SC_RICHMANKIM]->val1;
+			// jexp is boosted by both buff_ration AND buff_job_ratio, so we should not use buff_job_ratio here
```

Comment:
@skyleo can you check if this part about buff_ratio / buff_job_ratio is correct on the general exp calc? I find it very weird that job exp boost is buff_ratio + buff_job_ratio: (near the end of the function)

```
	bexp += apply_percentrate64(bexp, buff_ratio, 100);
	jexp += apply_percentrate64(jexp, buff_ratio + buff_job_ratio, 100);
```

maybe @hemagx remembers something... but this change is from 8 years ago 😅

Reply:
A bit odd since this means the bonuses from SC_CASH_PLUSEXP and SC_OVERLAPEXPUP are applied twice for job exp.

Also since when does SC_RICHMANKIM affect players? The experience calculation for SC_RICHMANKIM is in mob.c and basically only mobs that are inside Richmankim, will get SC_RICHMANKIM on them and thus give more experience.

From briefly reading iro wiki it seems that this hasn't changed. I suggest you introduce your change in mob.c at the Experience Calculation around line 2580 in function mob_dead.

Reply:
Yeah, I found odd that SC_CASH_PLUSEXP and SC_OVERLAPEXPUP applies twice, I did not test it in-game to check, but looks like a bug to me... not exactly a re-balance issue though.

---

Regarding RICHMANKIM:

After Renewal re-balance, SC_RICHMANKIM is a player buff just like Increase AGI/etc, there is no longer ground effects (monsters does not get SC_RICHMANKIM now).

From my in-game tests, the EXP is applied per-player.

Consider the following examples for a party of 2 players with even sharing, killing a monster where penalty is applied equally to both members (to make explanation simpler)

**Example 1:** Player A and Player B **NOT** buffed
    - Result: Player A and Player B gets the same amount of EXP

**Example 2:** Player A **Buffed** / Player B **NOT** buffed
    - Result: Player A always gets more EXP than Player B (regardless of who performed the kill)

**Example 3:** Player A and Player B **Buffed**
    - Result: Player A and Player B gets the same amount of EXP (increased by the effect)

And MVP exp does not get boosted.

I did try to keep it in mob.c, since we are talking about mob exp, but doing it there is... complicated. EXP at `mob_dead` did not consider the party EXP split yet, and since my tests have shown that a non-buffed player will get less EXP, I would need to change the party sharing code.